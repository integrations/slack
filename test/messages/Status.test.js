const { Status } = require('../../lib/messages/status');
const combinedStatus = require('../fixtures/combined_status.json');

describe('Status rendering', () => {
  test('works', async () => {
    const singleStatus = combinedStatus.statuses[1];
    const statusAttachment = new Status(singleStatus);
    const rendered = statusAttachment.renderAttachment();
    expect(rendered).toMatchSnapshot();
  });
  test('works for all checks pass', async () => {
    const rendered = Status.getChecksPassAttachment(5, 5);
    expect(rendered).toMatchSnapshot();
  });

  test('works for some checks pass', async () => {
    const rendered = Status.getChecksPassAttachment(4, 5);
    expect(rendered).toMatchSnapshot();
  });

  test('works when no checks pass', async () => {
    const rendered = Status.getChecksPassAttachment(0, 5);
    expect(rendered).toMatchSnapshot();
  });
});
