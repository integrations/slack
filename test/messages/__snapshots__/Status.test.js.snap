// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status rendering works 1`] = `
Object {
  "author_icon": "https://avatars2.githubusercontent.com/oa/1508?v=4",
  "author_link": "https://travis-ci.org/github-slack/app/builds/278661144?utm_source=github_status&utm_medium=notification",
  "author_name": "continuous-integration/travis-ci/push: The Travis CI build passed",
  "color": "#28a745",
  "fallback": "continuous-integration/travis-ci/push: The Travis CI build passed",
  "mrkdwn_in": Array [
    "text",
  ],
}
`;

exports[`Status rendering works for all checks pass 1`] = `
Object {
  "color": "#28a745",
  "fallback": "All checks have passed",
  "footer": "5/5 successful checks",
  "text": ":white_check_mark: All checks have passed",
}
`;

exports[`Status rendering works for some checks pass 1`] = `
Object {
  "color": "#dbab09",
  "fallback": "4 other checks have passed",
  "footer": "4/5 successful checks",
  "text": ":white_check_mark: 4 other checks have passed",
}
`;

exports[`Status rendering works when no checks pass 1`] = `
Object {
  "color": "#cb2431",
  "fallback": "No checks have passed",
  "footer": "0/5 successful checks",
  "text": ":white_check_mark: No checks have passed",
}
`;
